src/os/bluestore/bluestore_types.cc

```diff
@@ -833,6 +838,15 @@ void bluestore_onode_t::generate_test_instances(list<bluestore_onode_t*>& o)
   // FIXME
 }
 
+size_t bluestore_onode_t::get_preferred_csum_order() const
+{
+  uint32_t t = expected_write_size;
+  if (!t) {
+    return 0;
+  }
+  return ctz(expected_write_size);
```

Comment:
I think ctz so that if they specify a non-power of 2 (say 12K) we will csum at 4K.